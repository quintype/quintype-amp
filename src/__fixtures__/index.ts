import { config, publisherConfig, ampConfig, configOpts } from "./config.fixture";
import { textStory } from "./text-story.fixture";
import { storyWithManyJsEmbeds } from "./story-with-jsembeds.fixture";
import { relatedStories } from "./related-stories";
import { seo } from "./seo.fixture";
import { allElementsStory } from "./all-element-story.fixture";
import { liveBlogStory } from "./live-blog-story.fixture";

export {
  allElementsStory,
  config,
  publisherConfig,
  ampConfig,
  textStory,
  storyWithManyJsEmbeds,
  relatedStories,
  configOpts,
  seo,
  liveBlogStory
};
