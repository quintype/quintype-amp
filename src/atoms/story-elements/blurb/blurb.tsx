import React from "react";
import { StoryElementProps } from "../types";
import styled from "styled-components";
import { CommonRenderPropTypes } from "../../../types/config";
import { withStoryAndConfig } from "../../../context";

const StyledBlurb = styled.blockquote.attrs(({ style }: StoryElementProps & { style?: object }) => ({
  style: style
}))`
  font-size: ${(props) => props.theme.font.size.m};
  color: ${(props) => props.theme.color.mono7};
  margin: 0 ${(props) => props.theme.spacing.s};
  line-height: ${(props) => props.theme.font.lineHeight.level3};
  border-left: 3px solid ${(props) => props.theme.color.mono7};
  padding: 0 ${(props) => props.theme.spacing.m};

  // Applies when rendering direct html, fallback case
  blockquote {
    margin: 0 ${(props) => props.theme.spacing.s};
  }
`;

export const DefaultBlurb = ({ element, inlineStyles }: StoryElementProps) => {
  if (element.metadata && element.metadata.content) {
    return <StyledBlurb style={inlineStyles}>{element.metadata.content}</StyledBlurb>;
  }
  return <StyledBlurb style={inlineStyles} as="div" dangerouslySetInnerHTML={{ __html: element.text || "" }} />;
};

export const BlurbBase = ({ element, story, config }: StoryElementProps & CommonRenderPropTypes) => {
  return config.opts && config.opts.storyElementRender ? (
    config.opts.storyElementRender({ story, config })
  ) : (
    <DefaultBlurb element={element} story={story} config={config} />
  );
};

export const Blurb = withStoryAndConfig(BlurbBase);
