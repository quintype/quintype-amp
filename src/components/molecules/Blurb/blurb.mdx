---
name: Blurb
menu: Story Elements
route: "/story-elements/blurb"
---

### This is a blurb
