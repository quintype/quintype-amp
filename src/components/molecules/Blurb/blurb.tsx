import React from 'react'
import { Blurb } from './types'
import { AppContext } from '../../../helper-functions'
import styled from 'styled-components'

const StyledBlurb = styled.blockquote`
  color: red;
  background-color: purple;
`
// const StyledBlurb2 = styled.blockquote`
//   color: green;
//   background-color: grey;
// `

const AmpBlurb = (props: Blurb) => (
  <AppContext.Consumer>
    {appContext => {
      return appContext.ignoreDefaultStyles ?
        <blockquote className="amp-se-blurb">{props.children}</blockquote> :
        <StyledBlurb>{props.children}</StyledBlurb>
    }}
  </AppContext.Consumer>
)

export default AmpBlurb
