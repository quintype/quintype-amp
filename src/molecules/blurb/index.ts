import AmpBlurb from "./blurb";
export default AmpBlurb;
