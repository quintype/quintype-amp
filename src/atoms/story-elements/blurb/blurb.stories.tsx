import * as React from "react";
import { storiesOf } from "@storybook/react";
import { Blurb } from "./blurb";
import { config, textStory } from "../../../__fixtures__";
import { Layout } from "../../layout";

const sampleBlurbElement = {
  description: "",
  "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/19d96101-04b2-400a-ab95-e7362280d8a1",
  type: "text",
  "family-id": "5f761f27-4721-459f-8ecd-733021d8d2ea",
  title: "",
  id: "19d96101-04b2-400a-ab95-e7362280d8a1",
  metadata: {
    content:
      "In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest."
  },
  subtype: "blurb",
  text:
    "<blockquote>In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest.</blockquote>"
};

const { metadata, ...sampleBlurbElementWithoutMetadata } = sampleBlurbElement;

storiesOf("Blurb", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Default blurb", () => <Blurb element={sampleBlurbElement} />)
  .add("Blurb without metadata", () => <Blurb element={sampleBlurbElementWithoutMetadata} />)
  .add("Blurb with children", () => {
    return (
      // <Blurb>
      <Blurb element={sampleBlurbElement}>
        <div>
          <h1>Lorem Ipsum</h1>
        </div>
        <div>Hello World</div>
      </Blurb>
    );
  })
  .add("Blurb with custom styles", () => {
    const customStyles: any = {};
    customStyles.blurb = (theme) => ({
      color: `${theme.color.accent1}`,
      "font-size": `${theme.font.size.m}`
    });
    return <Blurb element={sampleBlurbElement} style={customStyles} />;
  })
  .add("Blurb with custom styles and children", () => {
    const customStyles: any = {};
    customStyles.blurb = (theme) => ({
      color: `${theme.color.accent1}`,
      "font-size": `${theme.font.size.m}`
    });
    return (
      <Blurb element={sampleBlurbElement} style={customStyles}>
        <div>
          <h1>Lorem Ipsum</h1>
        </div>
        <div>Hello World</div>
      </Blurb>
    );
  })
  .add("Blurb with custom styles and custom theme", () => {
    const customStyles: any = {};
    customStyles.blurb = (theme) => ({
      color: `${theme.color.accent1}`,
      "font-size": `${theme.font.size.m}`
    });
    return (
      <Blurb
        element={sampleBlurbElement}
        style={customStyles}
        theme={{
          color: {
            accent1: "orange"
          },
          font: {
            size: {
              m: "32px"
            }
          }
        }}
      />
    );
  });
