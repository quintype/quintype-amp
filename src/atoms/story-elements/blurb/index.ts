import { Blurb } from "./blurb";
export { Blurb };
