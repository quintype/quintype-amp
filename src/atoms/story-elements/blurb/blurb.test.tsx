import React from "react";
import { BlurbBase, DefaultBlurb } from "./blurb";
import { shallow } from "enzyme";
import { config, textStory } from "../../../__fixtures__";

const sampleBlurbElement = {
  description: "",
  "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/19d96101-04b2-400a-ab95-e7362280d8a1",
  type: "text",
  "family-id": "5f761f27-4721-459f-8ecd-733021d8d2ea",
  title: "",
  id: "19d96101-04b2-400a-ab95-e7362280d8a1",
  metadata: {
    content:
      "In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest."
  },
  subtype: "blurb",
  text:
    "<blockquote>In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest.</blockquote>"
};

const { metadata, ...sampleBlurbElementWithoutMetadata } = sampleBlurbElement;

describe("Blurb", () => {
  it("should render default", () => {
    const wrapper = shallow(<DefaultBlurb element={sampleBlurbElement} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render without metadata", () => {
    const wrapper = shallow(<DefaultBlurb element={sampleBlurbElementWithoutMetadata} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should call blurbRender prop when passed to opts", () => {
    const blurbRender = jest.fn();
    const modifiedConfig = { ...config, opts: { render: { storyElementRender: { blurbRender } } } };
    const wrapper = shallow(<BlurbBase element={sampleBlurbElement} story={textStory} config={modifiedConfig} />);
    expect(blurbRender.mock.calls.length).toBe(1);
    expect(wrapper.find(DefaultBlurb).length).toBe(0);
  });
});
