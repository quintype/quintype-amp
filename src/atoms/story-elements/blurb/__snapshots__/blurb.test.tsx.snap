// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blurb should render default 1`] = `
<styled.blockquote>
  In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest.
</styled.blockquote>
`;

exports[`Blurb should render without metadata 1`] = `
<styled.blockquote
  as="div"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<blockquote>In general—and this is a good P.S.A. reminder for everyone—if you are symptomatic, you are supposed to stay home for seven days, or three days past whenever your symptoms resolve, whichever one of those is longest.</blockquote>",
    }
  }
/>
`;
